qutebrowser/components/adblock.py

```diff
@@ -64,9 +64,20 @@ def get_fileobj(byte_io: typing.IO[bytes]) -> typing.IO[bytes]:
 
 def _is_whitelisted_url(url: QUrl) -> bool:
     """Check if the given URL is on the adblock whitelist."""
-    for pattern in config.val.content.host_blocking.whitelist:
+    for pattern in config.val.content.blocking.whitelist:
         if pattern.matches(url):
             return True
+
+    # Prevent a block-list from being able to block itself, otherwise you
+    # couldn't update it.
```

Comment:
Do you think this is still needed after the underlying bug was fixed in ab418af156b0cecc75e1f173af0096989ed97020? I suppose it's very unlikely to happen, and if it does, the user could always disable blocking manually.

Reply:
That's true. The run-time and maintenance costs of this extra check may not be worth the benefit it provides. I think I'll remove it.